### PR TITLE
Handle plain files as arguments: .info & .json only

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: src/producer.rs [hunk 1]
```diff
@@ -218,6 +235,13 @@ impl Archive {
                     true
                 }
                 Err(_) => false,
+            }
+            ArchiveType::Plain(_) => match File::open(name) {
+                Ok(mut f) => {
+                    f.read_to_end(buf).expect("Failed to read gcda file");
```
Comment (marco-c) [inline src/producer.rs]: Nit: this is not only used for `gcda` files, right?

File: src/producer.rs [hunk 2]
```diff
@@ -431,13 +471,31 @@ pub fn producer(
...
+                });
+            } else if let Some(ext) = full_path.clone().extension() {
+                let ext = ext.to_str().unwrap();
+                if ext == "info" || ext == "json" {
+                    plain_files.push(full_path);
+                } else {
+                    panic!("Cannot load file '{:?}': it isn't a .info or .json file.", full_path);
+                }
+            } else {
+                panic!("Cannot load file '{:?}': it isn't a .info or .json file.", full_path);
```
Comment (marco-c) [inline src/producer.rs]: Nit: `it isn't a directory, or a .info or a .json file`

File: src/producer.rs [hunk 3]
```diff
@@ -196,6 +197,22 @@ impl Archive {
                     }
                 }
             }
+            ArchiveType::Plain(ref plain) => {
+                // All the paths are absolutes
+                for full_path in plain {
+                    //let path = PathBuf::from(full_path.file_name().unwrap());
```
Comment (marco-c) [inline src/producer.rs]: Nit: remove commented out statement

File: src/producer.rs [hunk 4]
```diff
@@ -260,6 +284,20 @@ impl Archive {
 
                 true
             }
+            ArchiveType::Plain(_) => {
+                // don't use a hard link here because it can fail when src and dst are not on the same device
```
Comment (marco-c) [inline src/producer.rs]: We should never be in this branch, given that Plain is only for info files, right? If so, just remove all this code and add an assertion failing here

File: src/producer.rs [hunk 5]
```diff
@@ -431,13 +471,31 @@ pub fn producer(
...
-                item: RefCell::new(ArchiveType::Dir(full_path)),
-            });
+            if full_path.is_dir() {
+                archives.push(Archive {
+                    name: path.to_string(),
+                    item: RefCell::new(ArchiveType::Dir(full_path)),
+                });
+            } else if let Some(ext) = full_path.clone().extension() {
+                let ext = ext.to_str().unwrap();
+                if ext == "info" || ext == "json" {
```
Comment (marco-c) [inline src/producer.rs]: We could handle xml files too

File: src/producer.rs [hunk 6]
```diff
@@ -1412,6 +1470,29 @@ mod tests {
...
+    #[test]
+    fn test_plain_producer() {
+        let queue: Arc<WorkQueue> = Arc::new(MsQueue::new());
+
+        let tmp_dir = TempDir::new("grcov").expect("Failed to create temporary directory");
+        let tmp_path = tmp_dir.path().to_owned();
+        producer(
+            &tmp_path,
+            &[
+                "test/prova.info".to_string(),
```
Comment (marco-c) [inline src/producer.rs]: We should maybe test with a json and a xml too.

File: src/producer.rs [hunk 7]
```diff
@@ -1412,6 +1470,29 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_plain_producer() {
```
Comment (marco-c) [inline src/producer.rs]: Also test the case where we have a gcno and a gcda and check that the function panics

File: src/producer.rs (hunk 5 above)
Comment (calixteman) [reply src/producer.rs]: we could have several jacoco files coming from different machines

**Pull request merged**